### PR TITLE
Feature/improve client libs

### PR DESCRIPTION
**Pull request by woohsian**
This feature introduces automatic importing of the client lib files into the final HTML without touching the HTL script. There is more clear separation between the HTL and other files.

It also adds a master CSS and master JS file into the public (dist) folder as requested by Adobe.

Also fix regeneration of the components in webpack watch which was broken previously.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -57,8 +61,33 @@ const engine = require('./tools/main'),
...
             console.log("'%s' is a template to be processed.", filename);
             let ret = await engine(resource, template, resourceFile);
-            const filenameOut = path.resolve(process.cwd(), './generated_html/' + fileshorthtml);
-            fse.writeFile(filenameOut, ret.body, 'utf-8');
+            const filenameOut = path.resolve(process.cwd(), './generated_html/' + fileshorthtml),
+
+              dom = new JSDOM(ret.body);
+
+            for (let componentName in global.fullObj) {
+              let style = dom.window.document.createElement('link'),
```
Comment (ettiennegous) [inline index.js]: Very big method, please split it up a bit, you may be able to pull the content of the foreach into a nicely named method to make it clearer what its doing

File: test/components/info/clientlibs/js/info-js.js [hunk 2]
```diff
@@ -0,0 +1 @@
+(function(a){function b(d){if(c[d])return c[d].exports;var e=c[d]={i:d,l:!1,exports:{}};return a[d].call(e.exports,e,e.exports,b),e.l=!0,e.exports}var c={};return b.m=a,b.c=c,b.d=function(a,c,d){b.o(a,c)||Object.defineProperty(a,c,{enumerable:!0,get:d})},b.r=function(a){'undefined'!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(a,Symbol.toStringTag,{value:'Module'}),Object.defineProperty(a,'__esModule',{value:!0})},b.t=function(a,c){if(1&c&&(a=b(a)),8&c)return a;if(4&c&&'object'==typeof a&&a&&a.__esModule)return a;var d=Object.create(null);if(b.r(d),Object.defineProperty(d,'default',{enumerable:!0,value:a}),2&c&&'string'!=typeof a)for(var e in a)b.d(d,e,function(b){return a[b]}.bind(null,e));return d},b.n=function(a){var c=a&&a.__esModule?function(){return a['default']}:function(){return a};return b.d(c,'a',c),c},b.o=function(a,b){return Object.prototype.hasOwnProperty.call(a,b)},b.p='',b(b.s=80)})({80:function(){console.log('This is component info loaded')}});
```
Comment (ettiennegous) [inline test/components/info/clientlibs/js/info-js.js]: Is this compiled code? Should it be added to the git ignore and deleted?

File: index.js (hunk 1 above)
Comment (woohsian) [reply index.js]: Refactored this

File: test/components/info/clientlibs/js/info-js.js (hunk 2 above)
Comment (woohsian) [reply test/components/info/clientlibs/js/info-js.js]: I have removed this from the repo

**Review by ettiennegous — APPROVED** [review]

**Pull request merged**